packages/core/src/domain/configuration/configuration.ts

```diff
@@ -95,12 +97,44 @@ export type Configuration = typeof DEFAULT_CONFIGURATION &
   TransportConfiguration & {
     cookieOptions: CookieOptions
 
-    service?: string
-    beforeSend?: BeforeSendCallback
+    service: string | undefined
```

Comment:
Does it change something?
Should we do the same for `actionNameAttribute`?

Reply:
> Does it change something?

Yes, it ensures that the property is set. For example:

```ts
type Configuration = { foo?: string }

let configuration: Configuration = {} // no error. We may have forgot to set `foo` from `initConfiguration`


type Configuration = { foo: string | undefined }

let configuration: Configuration = {} // error
let configuration: Configuration = { foo: initConfiguration.foo } // ok
```

> Should we do the same for `actionNameAttribute`?

This will be changed in the RUM PR. This is a bit weird to change it now, since we don't want to set it in the core `validateAndBuildConfiguration` function.

Reply:
OK, I think I get it, the idea is to be sure that for the internal configuration object we set all the needed fields even if they are not set in the customer facing configuration.